services/web/server/src/simcore_service_webserver/workspaces/_models.py

```diff
@@ -0,0 +1,77 @@
+import logging
+
+from models_library.basic_types import IDStr
+from models_library.rest_base import RequestParameters, StrictRequestParameters
+from models_library.rest_filters import Filters, FiltersQueryParameters
+from models_library.rest_ordering import (
+    OrderBy,
+    OrderDirection,
+    create_ordering_query_model_classes,
+)
+from models_library.rest_pagination import PageQueryParameters
+from models_library.trash import RemoveQueryParams
+from models_library.users import GroupID, UserID
+from models_library.workspaces import WorkspaceID
+from pydantic import BaseModel, Extra, Field
+from servicelib.request_keys import RQT_USERID_KEY
+
+from .._constants import RQ_PRODUCT_KEY
+
+_logger = logging.getLogger(__name__)
+
+
+class WorkspacesRequestContext(RequestParameters):
+    user_id: UserID = Field(..., alias=RQT_USERID_KEY)  # type: ignore[literal-required]
+    product_name: str = Field(..., alias=RQ_PRODUCT_KEY)  # type: ignore[literal-required]
+
+
+class WorkspacesPathParams(StrictRequestParameters):
+    workspace_id: WorkspaceID
+
+
+_WorkspacesListOrderQueryParams: type[
+    RequestParameters
+] = create_ordering_query_model_classes(
+    ordering_fields={
+        "modified_at",
+        "name",
+    },
+    default=OrderBy(field=IDStr("modified_at"), direction=OrderDirection.DESC),
+    ordering_fields_api_to_column_map={"modified_at": "modified"},
+)
+
+
+class WorkspacesFilters(Filters):
+    trashed: bool | None = Field(
+        default=False,
+        description="Set to true to list trashed, false to list non-trashed (default), None to list all",
+    )
+
+
+class WorkspacesListQueryParams(
+    PageQueryParameters,
+    FiltersQueryParameters[WorkspacesFilters],
+    _WorkspacesListOrderQueryParams,  # type: ignore[misc, valid-type]
+):
+    ...
+
+
+class WorkspacesGroupsPathParams(BaseModel):
+    workspace_id: WorkspaceID
+    group_id: GroupID
+
+    class Config:
+        extra = Extra.forbid
+
+
+class WorkspacesGroupsBodyParams(BaseModel):
+    read: bool
+    write: bool
+    delete: bool
+
+    class Config:
+        extra = Extra.forbid
```

Comment:
question: for the path it kind of makes sense to have forbid. for the body I am wondering. if the client puts more stuff we do not care right? as long as we want is there. of course it is easier to debug the client if it passes more stuff... well.. anyway. just a question.

Reply:
I find it handy to be strict. Otherwise, on the client side, you will have a headache debugging why it isn't working as expected.

Reply:
That is indeed very important for the path parameters. E.g. in fastapi, the order of registration of the router matter so it is fairly common to get the requests in the wrong entrypoint. These validator is a safe measure

e.g. if these routes `GET parent/{parent_id}/item/{item_id}` and `GET parent/{parent_id}/item/latest`  are added wrong, the requests intended for the first might end up in the second